src/lib/urlPatterns.ts

```diff
@@ -13,6 +14,24 @@ export function extractYouTubeVideoId(url: string) {
   return searchParams.get('v')
 }
 
+export function extractYouTubePlaylistParams(url: string) {
+  const urlObj = new URL(url)
+  const searchParams = new URLSearchParams(urlObj.search)
+  return {
+    list: searchParams.get('list'),
+    index: searchParams.get('index'),
+  }
```

Comment:
```suggestion
  return Object.fromEntries(urlObj.searchParams.entries())
```

If changed, I'd rename the function to something like `extractYouTubeSearchParams`.